addons/udes_sale_stock/tests/test_sale_order.py

```diff
@@ -63,7 +65,16 @@ def test_only_moves_created_by_sale_order_are_cancelled(self):
 
         second_sale_order.action_cancel()
         self.assertEqual(second_move.state, "cancel")
-    
+
+    @freezegun.freeze_time("2025-04-28 15:30:00", tz_offset=0)
+    def test_done_date_set_when_sale_marked_done(self):
+        """Test that u_done_date is set when Sale Order is marked as done and is close to now."""
+        sale_order = self._create_sale_order(self.apple.id, 10)
+        sale_order.action_confirm()
+        self.assertFalse(sale_order.u_done_date, "Done Date should not be set initially.")
+        sale_order.action_done()
+        self.assertEqual(sale_order.u_done_date, datetime.strptime("2025-04-28 00:00:00", "%Y-%m-%d %H:%M:%S"))
```

Comment:
I think this assertion will fail and needs to be changed
```suggestion
        self.assertEqual(sale_order.u_done_date, datetime.strptime("2025-04-28 15:30:00", "%Y-%m-%d %H:%M:%S"))
```